src/java.base/share/classes/java/util/concurrent/LinkedBlockingDeque.java

```diff
@@ -1342,15 +1344,15 @@ public boolean removeIf(Predicate<? super E> filter) {
     /**
      * @throws NullPointerException {@inheritDoc}
      */
-    public boolean removeAll(Collection<? extends @NonNull Object> c) {
+    public boolean removeAll(Collection<?> c) {
         Objects.requireNonNull(c);
         return bulkRemove(e -> c.contains(e));
     }
 
     /**
      * @throws NullPointerException {@inheritDoc}
      */
-    public boolean retainAll(Collection<? extends @NonNull Object> c) {
+    public boolean retainAll(Collection<?> c) {
```

Comment:
What is the motivation for this and some similar changes?
The javadoc for the NPE in `Collection` says `NullPointerException if this collection contains one or more null elements and the specified collection does not permit null elements`.
As these blocking queues don't allow null, why allow passing a collection that can contain null?

Reply:
Is the point that these are covariant changes to the signature from `Collection` and therefore unsound?

Reply:
> As these blocking queues don't allow null, why allow passing a collection that can contain null?

These changes aren't prompted by any specific need. I took the overall principle to be "accept the most general type that can't lead to NPE," so I generalized the signature. (This is similar to how methods like `contains` are currently declared to accept a `@Nullable` input—though I admit to being slightly surprised by that, given that the docs say that `null` is supported only very indirectly :)) I agree that it's unlikely that anyone would want to pass a collection that actually contains nulls. It's maybe more believable that someone would pass a collection that the checker _can't prove_ will never contain nulls. But even in that case, I'm not specifically aware of a need.

I don't mind reverting these to get the rest of the PR (mainly, the `@Nullable` returns) merged if that's less controversial. Let me know.

Reply:
I've created https://github.com/eisop/checker-framework/issues/232 to think about how to handle this properly. I'm fine with removing the stricter bounds for now.